storm-core/src/jvm/org/apache/storm/topology/TopologyBuilder.java

```diff
@@ -113,6 +113,15 @@ public StormTopology createTopology() {
         Map<String, Bolt> boltSpecs = new HashMap<>();
         Map<String, SpoutSpec> spoutSpecs = new HashMap<>();
         maybeAddCheckpointSpout();
+
+        if (_spouts.size() == 0) {
+            throw new IllegalArgumentException("Spouts is not set");
+        }
+
+        if (_bolts.size() == 0) {
```

Comment:
We have topologies that only have spouts (no bolts).  I know it is kind of an odd corner case where someone is just reading from an input and writing to an output in one shot, but it is something that we have.